decisions/websre/0005-web-sre-kubernetes-secrets-pattern.md

```diff
@@ -0,0 +1,55 @@
+
+There are currently a few ways the secrets can be written to the cloud secret store. We have some terraform that writes the secrets directly to the secrets stores, but most secrets are populated manually by operators.  We expect both methods to be used going forward.  In general, we expect SREs/Operators with elevated privileges to manage the secrets.  This ADR intentionally doesn't include scope for devs to write new secrets, or update existing ones.
+
+The default assumption for storage patterns in aws secrets store is that you follow the pattern: `<group>/<app>/<environment>/<secretname>` for example `mozmeao/snippets/stage/database_url`  This seems to make writing aws iam access stanzas easy.  Not needing to be rewritten with each new secret added.  But, enough separation to prevent us from accidentally crossing wires.
+
+We want to choose just one secret storage and retrieval method, because having one known good way makes operators experience much simpler.  The alternatives/already existing methods are listed below with a quick note on why we didn't standardize on them.
+
+We are choosing to not use a few other types of secret storage mechanisms, including (but possibly not limited to):
+* git-crypt secret repos + sops, to k8s secrets directly.
+  * Many manual steps to get secrets to prod
+  * Difficult to remove access after people left (git-crypt doesn't support removing keys, you have to re-encrypt)
+* Sealed secrets
+  * The workflow and pattern is similar to externalsecrets, keeping both therefore doesn't make sense (doubles our maintenance burden, with little extra benefit)
+  *  sealed secrets require extra cli tools and setup on operator computers, so is just ever so slightly more difficult to use than externalsecrets
+* SSM Parameter Store (lambdas/codebuilds use this)
+  * is mostly just an earlier version of secrets manager
+  * splitting secrets between two places leads to duplication
+  * the workflows are so similar maintaining both seems sub-optimal
+* no new systems to be built, for example hashicorp vault
+  * that's an entire project in and of itself, so we're choosing to not do that because the 'buy' choice is good enough.
+* adding secrets directly to docker images, or directly in source code (high priority to fix these!)
+  * Avoiding these because security has informed us this is explicitly a bad practice. Makes it way too easy to leak the secret to the world.
+
+
+
+We expect CI/CD pipelines to use their current secret storage, and not try to connect them to the secret store.  The one exception is the cloud native pipeline solutions (e.g AWS CodeBuild), which should use the secrets manager if possible.  We may end up duplicating secrets, but in general we should try to keep application secrets and pipeline secrets separate, with pipeline secrets being manually added in the right context.
+
+## Consequences
+
+- We will be migrating secrets, which is a reasonably risky process for many applications.
```

Comment:
I could read this sentence as either setting the expectation that we will migrate all secrets to external secrets, or as just calling out that when we do migrate secrets it will pose some risk.

My expectation is that this ADR commits us to using external secrets when there is an opportune moment to rework an apps secrets handling (e.g. the cluster migration work going on now, or moving sumo to eks) or when we are launching an app on k8s (e.g. a totally new app like ctms, or moving something currently non-k8s like wikimo). I don't think it commits us to migrating all secrets for all apps.